channel/state/.adr/0000-signature-scheme.md

```diff
@@ -0,0 +1,15 @@
+## 0000 -- State Signature Scheme
+
+Our state signing scheme involves encoding a data structure into a string of bytes, hashing those and signing the hash. Signing a hash is important because it protects against leakage of the private key -- since the hash is unpredicatble it cannot be crafted maliciously by an adversary (even if the preimage has been prepared by that adversary).
+
+There remains a potential issue with signatures being resued in unintended ways.
+
+Consider an encoding collision p = encode(a,b) = encode'(a',b') = p'. Put otherwise, the bytes p could be encoded / decoded in two different ways (perhaps the intended way plus another unintended way), giving rise to distinct variables a' != a, b' != b. If these distinct sets of variables and encoding functions result in the same preimage for the hashing and signing steps, the same signature could be "valid" for both sets.
+
+We address this issue as follows:
+
+Nitro clients should verify the decoding from the contract where funds are locked. Notwithstanding things like upgradeable contracts, if we use `abi.encode` this decoding is deterministic and therefore totally predictable. (Contrast this with using something like `abi.encodePacked`, where multiple inputs encode to the same bytes -- as a consequence there is no decoder for `abi.encodePacked` as the there is no unique pre-image).
+
+A signature on `h(p)` might get "replayed" to a contract with a different decoder, with unintended effects -- but as long as no funds are ever sent there, there should be no issue. We do need to ensure that the signature is generated by a truly ephemeral key, though -- the only place where that key should have meaning is for signing nitro state channel updates.
```

Comment:
> We do need to ensure that the signature is generated by a truly ephemeral key, though -- the only place where that key should have meaning is for signing nitro state channel updates.

It's not totally obvious to me why this is true. I'll take a crack at it:

_If a key `k` is used for signing state channel updates as well as commitments in some other protocol `P`, then there's a risk that the designer of protocol `P` designed the possibility of a hash collision **by means of a specially crafted encoder.**_

Is that about right?

Reply:
For each signature made with a given private key, the owner of said key should be checking on the consequences of that signature for each and every contract (on each and every chain) which holds funds or other important state for the owner and can be controlled to some extent by that key. This includes the native assets and accounts of those chains. They do these checks on the data to be signed, and decline to sign if they spot an issue. 

Such contracts/chains might only be deployed _after_ some signatures have been made with that key. So the owner of the key needs to also check all previous signatures that have been made with that key against the new chain/contract before depositing funds there. This is pretty infeasible, too, so the owner should select a new private key for every new protocol that they enter* in order to remain safe.

An example attack would be to take the data which was signed during execution of an existing protocol (for example, state channels) and convince the signer to partake in a new protocol backed by a contract which has the "specially crafted encoder" that you mention. If the signer deposits money it could be at risk by having the existing signatures (which are already in the wild) played onto the new contract. 

---
*this reminds me of the conversation in standup about enter/exit

Reply:
Can we find existing literature that covers this nuance? If so, we should link to it somehow. If not, it deserves to be written.